lib/gollum-lib/wiki.rb

```diff
@@ -510,10 +510,8 @@ def revert_page(page, sha1, sha2 = nil, commit = {})
         sha2   = nil
       end
 
-      patch     = full_reverse_diff_for(page, sha1, sha2)
       committer = Committer.new(self, commit)
-      parent    = committer.parents[0]
-      committer.options[:tree] = @repo.git.apply_patch(parent.sha, patch)
+      committer.options[:tree] = @repo.git.revert(page ? page.path : nil, sha1, sha2, committer.parents[0])
```

Comment:
I'm not sure I understand this behavior here, as the method is called `revert_page`. So I would assume that this method has to be called with page object. There is a separate `revert_commit` method below, which should do something else than revert_page.